src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -84,11 +85,25 @@ def get_default_kernel_args(self) -> List[str]:
         """
         raise NotImplementedError
 
-    @abstractmethod
     def get_disk_device(self) -> str:
         """
         Get the disk device, e.g., "/dev/sda", where the disk image is placed.
 
+        :returns: The disk device.
+        """
+        if self._disk_device is None:
+            warn("No disk device set, ie where the disk image is located. Defaulting to board disk device")
```

Comment:
I think a warning here is probably not necessary. Up until this PR using these defaults was normal behavior. How about we change this from a `warn` to an `inform` (`from m5.util import inform`). I'd simply state, for either case, 'Disk Device set to "{VALUE HERE}"'.